consensus/core/src/authority_service.rs

```diff
@@ -221,11 +221,22 @@ impl<C: CoreThreadDispatcher> NetworkService for AuthorityService<C> {
         &self,
         peer: AuthorityIndex,
         block_refs: Vec<BlockRef>,
+        highest_accepted_rounds: Vec<Round>,
     ) -> ConsensusResult<Vec<Bytes>> {
+        const MAX_ADDITIONAL_BLOCKS: usize = 10;
         if block_refs.len() > self.context.parameters.max_blocks_per_fetch {
             return Err(ConsensusError::TooManyFetchBlocksRequested(peer));
         }
 
+        if !highest_accepted_rounds.is_empty()
```

Comment:
nit: remove redundant check `!highest_accepted_rounds.is_empty()` as `highest_accepted_rounds.len()` will return 0 if it is empty

Reply:
The `!highest_accepted_rounds.is_empty()` allow us to make the vector optional. So the other peers can either provide an empty vector (thus will not attempt to provide back any additional blocks), or if they do provide a non-empty vector it will have to be the same size as the committee size.